### DISCUSSION TITLE
Gitter Alternatives -- Utilize stack overflow for user questions?

### DISCUSSION THREAD

**Discussion by magnunor** [hyperspy/hyperspy#3374] [Ideas]
Currently `gitter` is used to answer user questions, and we suggest people ask questions there.

While this works quite well, it has one downside: the answers aren't very "search-engine-visible", thus the `gitter` answers will be read by a very small number of people.

Would it be an idea try to utilize stack overflow for user questions? This way, the answers would be more visible on search engines.

I think the HyperSpy `gitter` would still be useful for certain kinds of questions and conversations, but the more "well-defined" questions could be answered at stack overflow.

One way facilitating this, would be to make a hyperspy tag: https://stackoverflow.com/help/tagging

**Comment by CSSFrancis** [comment]
Just a note that I think you need to have answered a certain amount of questions to make a tag. Not a huge problem but if no one here can do that we might have to ask someone to do it for us.

I do like the idea that it becomes much more Google friendly if we do this. But I also do wonder if the barriers to asking a question becomes a little higher. Then again I don't think that we are trying to completely remove Gitter so even if we can get some of the more common questions answered it would probably be good for the community.

**Comment by jlaehne** [comment]
Having a sort of searchable Q&A definitely sounds worthwile.

Searching stackoverflow for `hyperspy` gives already 13 mentions

**Comment by ericpre** [comment]
Indeed, this sounds sensible. Apparently, anyone can create but if it is not used in a 6 months period, it will deleted: https://stackoverflow.com/help/privileges/create-tags

**Comment by CSSFrancis** [comment]
This was brought up in the Hyperspy tutorial and I think people don't like the lack of a good serach functionality in gitter/element.  It at least deserves additional discussion. 

@M0hsend

**Comment by ericpre** [comment]
Regarding stackoverflow, does anyone who how can we get notification when there is a question related to hyperspy? I did answer some of the hyperspy question in the past (one answer was deleted to, no idea why!) but we don't proactively search for hyperspy question, I doubt that they will be answered... Also, it seems, that it there is a limitation in the number of answer you can post in a given time, which doesn't help...

I am wondering, it we should request hyperspy/pyxem/exspy/lumispy tags in https://discuss.scientific-python.org?
There are also https://forum.image.sc (very image focused) and https://matsci.org (oriented on material science modelling) but I don't think that these are not generic enough to be representive of hyperspy.

**Comment by ericpre** [comment]
What's about simply using GitHub Discussions, as for example in this case: https://github.com/hyperspy/hyperspy/discussions/3378?

It has all what is needed:
- it is easily searchable
- it is possible to pick an answer as resolved (similar as stack overflow), to avoid having to go through the whole discussion
- for people interested in following the discussion only, this is possible:
![image](https://github.com/hyperspy/hyperspy/assets/11851990/20000e27-5cf8-4ad0-9af3-31d7b72931ce)
- issue can very easily be created from a discussion
- more likely to get the attention of the developers! 😉

**Reply by CSSFrancis** [reply]
+1 for this idea.  I think it brings people to the github, that is where the developers are and it's checked daily.  We can also set up a template for questions:

Something like how to search for already asked questions etc.
